drivers/dai/intel/ssp/ssp.c

```diff
@@ -1986,14 +2058,21 @@ static void dai_ssp_set_reg_config(struct dai_intel_ssp *dp, const struct dai_co
 	sys_write32(sscr1, dai_base(dp) + SSCR1);
 	sys_write32(regs->ssc2 | SSCR2_SFRMEN, dai_base(dp) + SSCR2); /* hardware specific flow */
 	sys_write32(regs->ssc2, dai_base(dp) + SSCR2);
+#ifndef CONFIG_SOC_INTEL_ACE30_PTL
```

Comment:
I guess it's always hard to predict how future products will be but... are you sure about the negative `#ifndef ACE30_PTL` here? As opposed to a positive `#ifdef ACE20_LNL || ...`

"Everything but PTL" sounds... unusual: does PTL do something very unusual here? Sorry I'm not familiar with the hardware (someone familiar with the hardware should definitely review).

Reply:
I use the #ifndef PTL to simplify  struct definition. Without it, the fields should be created for all older platform than PTL.

Reply:
I understand, but what about future platforms? Won't their list grow even larger?

Reply:
To solve this problem of ever growing platform lists, https://github.com/thesofproject/sof/pull/9185/files#diff-1c06ec99653e34b03e0ed9fec285e348602417504608b2b2f1b38bd667434311 uses ACE_VERSION_XXX < ... < ACE_VERSION_YYY

Reply:
Ok, good idea - once the above change is merged (PTL upstream PR), I will prepare a new PR with the correction of the define and the addition of comments. Thanks.